example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,370 @@
+  Path _buildBorderPath(ArrowDirection arrowDirection, double arrowCenter) {
+    final halfOfBase = arrowBaseWidth / 2;
+
+    // Adjust the rect to leave space for the arrow.
+    // During layout, we reserve space for the arrow in both x and y axis.
+    final contentRect = Rect.fromLTWH(
+      arrowLength,
+      arrowLength,
+      size.width - arrowLength * 2,
+      size.height - arrowLength * 2,
+    );
+
+    Path path = Path()..addRRect(RRect.fromRectAndRadius(contentRect, Radius.circular(borderRadius)));
+
+    // Add the arrow.
+    if (arrowDirection == ArrowDirection.left) {
+      path
+        ..moveTo(contentRect.centerLeft.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.right) {
+      path
+        ..moveTo(contentRect.centerRight.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.up) {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.topCenter.dy)
+        ..relativeLineTo(halfOfBase, -arrowLength)
+        ..relativeLineTo(halfOfBase, arrowLength);
+    } else {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.bottomCenter.dy)
+        ..relativeLineTo(halfOfBase, arrowLength)
+        ..relativeLineTo(halfOfBase, -arrowLength);
+    }
+
+    path.close();
+
+    return path;
+  }
+
+  /// Computes the direction where the arrow should point to.
+  ///
+  /// If [globalFocalPoint] is inside the [menuRect] horizontal bounds, the arrow points up or right. Otherwise,
+  /// the arrow points left or right.
+  ///
+  /// If [allowHorizontalArrow] is `false`, the arrow only points up or down.
+  ArrowDirection _computeArrowDirection(Rect menuRect, Offset globalFocalPoint) {
+    final isFocalPointInsideHorizontalBounds =
+        globalFocalPoint.dx >= menuRect.left && globalFocalPoint.dx <= menuRect.right;
+    if (isFocalPointInsideHorizontalBounds || !allowHorizontalArrow) {
+      if (globalFocalPoint.dy < menuRect.top) {
+        return ArrowDirection.up;
+      }
+      return ArrowDirection.down;
+    } else {
+      if (globalFocalPoint.dx < menuRect.left) {
+        return ArrowDirection.left;
+      }
+      return ArrowDirection.right;
+    }
+  }
+
+  /// Computes the center point of the arrow.
+  ///
+  /// This point can be on the x or y axis, depending on the [direction].
+  double _computeArrowCenter(ArrowDirection direction, Offset focalPoint) {
+    final desiredFocalPoint = _isArrowVertical(direction) //
+        ? focalPoint.dx
+        : focalPoint.dy;
+
+    return _constrainFocalPoint(desiredFocalPoint, direction);
+  }
+
+  /// Computes the (x, y) offset used to paint the menu content inside the popover.
+  ///
+  /// When [direction] is up or left, the content needs to be shifted
```

Comment:
Why would we move the content for a particular arrow direction?

Reply:
It doesn't affect it anymore. I updated it now.